nucleus/security/core/src/main/java/com/sun/enterprise/security/PolicyLoader.java

```diff
@@ -125,13 +125,21 @@ public void loadPolicy() {
 
                 boolean usePolicyProxy = Boolean.parseBoolean(System.getProperty(POLICY_PROXY, "true"));
 
-                Policy policy = null;
+                Policy policy;
                 if (usePolicyProxy && System.getSecurityManager() != null) {
                     policy = loadPolicyAsProxy(javaPolicyClassName);
                 } else {
                     policy = loadPolicy(javaPolicyClassName);
                 }
 
+                // TODO: causing ClassCircularity error when SM ON and
```

Comment:
Do you know if the TODO is still true? In GF8 we remove everything related to SecurityManagers, so I would not spend much time with that, just if you noticed something related. The comment is there all the time GF moved to Eclipse.

Reply:
> Do you know if the TODO is still true?

I did not any tests.

Reply:
So, a simple app using application libraries and with SM enabled works without any problems